libgalois/include/katana/PropertyGraph.h

```diff
@@ -18,6 +18,21 @@
 
 namespace katana {
 
+// TODO(amber): find a better place to put this
+template <
+    typename T,
+    typename __Unused = std::enable_if_t<std::is_arithmetic<T>::value>>
+auto
+ProjectAsArrowArray(const T* buf, const size_t len) noexcept {
+  using ArrowDataType = typename arrow::CTypeTraits<T>::ArrowType;
+  using ArrowArrayType = arrow::NumericArray<ArrowDataType>;
+  return std::make_shared<ArrowArrayType>(len, arrow::Buffer::Wrap(buf, len));
+}
+
+/// Provides write access to private data of GraphTopology.
+/// WARNING: Dangerous
+class KATANA_EXPORT GraphTopologyAccess;
```

Comment:
Shouldn't `KATANA_EXPORT` be on the class definition (as well)?

Also you can consider not exporting the access class, e.g.,

```cpp

class GraphTopologyAccess;

... friend class GraphTopologyAccess ...

class KATANA_EXPORT MyAccess: public GraphTopologyAccess {

};

```

Reply:
Can you please explain the benefit? 
Related to this, I had a discussion with Roshan and Simon and this class may not be needed, so I might get rid of it.

Reply:
Removed GraphTopologyAccess in 6f4b8a0971e97a7fcad213f68d6e2d97359f777c